code_locator/retrieval/bm25s_client.py

```
@@ -131,6 +227,14 @@ def load(self, index_dir: str) -> None:
         self._doc_ids = data["doc_ids"]
         self._loaded = True
 
+        sym_path = Path(index_dir) / "bm25_symbol_index.pkl"
+        if sym_path.exists():
+            with open(sym_path, "rb") as f:
+                sym_data = pickle.load(f)
```

Ruff (0.15.10)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)